src/hangar/external/manage_plugins.py

```diff
@@ -0,0 +1,219 @@
+    plugin_details_store.clear()
+    plugin_provides.clear()
+    plugin_module_name.clear()
+    plugin_meta_data.clear()
+    _plugin_store.clear()
+
+
+def _scan_plugins():
+    """
+    Scan the plugins directory for .ini files and parse them
+    to autofill storage
+    """
+    pd = os.path.dirname(__file__)
+    config_files = glob(os.path.join(pd, '_plugins', '*.ini'))
+
+    for filename in config_files:
+        name, meta_data = _parse_config_file(filename)
+        plugin_meta_data[name] = meta_data
+
+        for a in meta_data['accepts'].split(','):
+            for p in meta_data['provides'].split(','):
+                accepts = a.strip()
+                provides = p.strip()
+                if provides not in valid_provides:
+                    warn(f"Plugin: {name} wants to provide non-existent: {p}. Ignoring.")
+                    continue
+                plugin_provides[name].append(provides)
+                tmp_acc = plugin_details_store.get(accepts)
+                if tmp_acc:
+                    tmp_prv = tmp_acc.get(provides)
+                    if tmp_prv:
+                        tmp_prv.append(name)
+                    else:
+                        tmp_acc[provides] = [name]
+                else:
+                    plugin_details_store[accepts] = {provides: [name]}
+        plugin_module_name[name] = os.path.basename(filename)[:-4]
+
+
+def _read_preferred_plugins_config():
+    """
+    Populate preferred plugin dict that maps file formats to plugins and methods. This
+    is used to infer which plugin to use at runtime based on file format
+    """
+    parser = configparser.ConfigParser()
+    config_path = os.path.join(os.path.dirname(__file__), 'preferred_plugins.ini')
+    parser.read(config_path)
+    return parser
+
+
+def _write_preferred_plugins_config(parser: configparser.ConfigParser):
+    """
+    Write current preferred plugin config object to file
+    """
+    config_path = os.path.join(os.path.dirname(__file__), 'preferred_plugins.ini')
+    with open(config_path, 'w') as f:
+        parser.write(f)
+
+
+def _import(plugin: str, method: str) -> Callable:
+    if plugin in _plugin_store:
+        loaded_plugin = _plugin_store[plugin]
+    else:
+        modname = plugin_module_name[plugin]
+        loaded_plugin = __import__(f'hangar.external._plugins.{modname}', fromlist=[modname])
+        _plugin_store[plugin] = loaded_plugin
+    return getattr(loaded_plugin, method)
+
+
+def _parse_config_file(filename: str) -> Tuple[str, dict]:
+    """Return plugin name and meta-data dict from plugin config file."""
+    parser = configparser.ConfigParser()
+    parser.read(filename)
+    name = parser.sections()[0]
+    meta_data = {}
+    for opt in parser.options(name):
+        meta_data[opt] = parser.get(name, opt)
+    return name, meta_data
+
+
+def _default_name_sampler(fpath):
+    return os.path.splitext(os.path.basename(fpath))[0]
+
+
+def reset_plugins():
+    global _preferred_plugins
```

Comment:
Does this actually need to be a global? or should we be using a `Plugin` class here?

Reply:
hmm, we could. Although I don't see any troubles by using global in this particular use case. It was essentially a clean up of what we had in the existing `manage_plugin.py`. I'll make a PluginManager class